crates/core/src/network/incoming.rs

```diff
@@ -620,7 +620,11 @@ impl<O: RpcResponse + WireEncode> Reciprocal<Oneshot<O>> {
 
     /// Fails if connection was already dropped
     pub fn try_send(self, msg: O) -> Result<(), ConnectionClosed> {
-        let reply = O::encode_to_bytes(&msg, self.protocol_version);
+        // The assumption here is that because this is a reply of an RPC request
+        // that we received on this connection, we hard-assume that the reply type
+        // serializable on the negotiated protocol version.
+        let reply = O::encode_to_bytes(&msg, self.protocol_version)
+            .expect("serialization of rpc reply should not fail");
```

Comment:
My immediate reaction was, is it worth panicking over this? But I think it makes sense how you have it. Simply moving on might be a correctness error in the worst case.

Reply:
if it's an invariant, I'd prefer to not silently fail whenever possible.